### ISSUE TITLE
Array pop does not change state

### ISSUE THREAD

**Issue by rmsm** [codecombat/esper.js#8]
`Array.pop` is no longer updating the state in the latest build.

Current build (28 September 2017, version ad69d01):
```
var testArray = esper.eval("var testArray = ['one', 'two', 'three']; testArray.pop(); testArray");
console.log('testArray', testArray); //  ['one', 'two', 'three']
```

The current version doesn't update the` testArray` variable after `pop`.

Previous build on Esper.js demo website (23 Aug 2017, unversioned):
```
var testArray = esper.eval("var testArray = ['one', 'two', 'three']; testArray.pop(); testArray");
console.log('testArray', testArray); //  ['one', 'two']
```

The previous build has the correct behavior.

**Comment by basicer** [comment]
Interesting.

The problem here is actually that esper is missing the magic behavior where setting the length value of an array truncates the integer key values.  This misbehavior was sometimes masked in older version when the object was copied out of the esper because it would set the length property on the output object, and the browser would apply the magic behavior.

**Comment by timert** [comment]
That's strange. Do you know why the unit tests are passing?

```
type-array.tests.js
        Array.prototype.pop
          ✓ should return the last item from the array
          ✓ should update the length
          ✓ should return undefined for an empty array
          ✓ should not affect length of empty array
```

**Comment by basicer** [comment]
That sorta works, but doesn't really fix the underlaying problem.  

```javascript
function p(o) { console.log(JSON.stringify(o)); }
var arr = [1,2,3];
p(arr);
arr.length = 1;
p(arr);
p(arr[2]);
```

```
Anya:~ rob$ node o.js 
[1,2,3]
[1]
undefined


Anya:~ rob$ esper o.js 
o.js
[1,2,3]
[1]
3
```

**Comment by basicer** [comment]
Should be fixed by 8e431a6
